src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +53,300 @@ const ActorInit En_Ani_InitVars = {
+        this->actor.gravity = 0.0f;
+        this->actor.flags |= ACTOR_FLAG_10;
+        this->stateFlags |= ANI_STATE_CLIMBING;
+        gSaveContext.eventInf[1] &= (u8)~0x10;
+
+    } else { // ANI_TYPE_STANDING
+        // for some reason standing he has a large collider
+        // possibly he was meant to be a blocking npc like bomber
+        this->collider2.dim.radius = 60;
+        this->actionFunc = EnAni_IdleStanding;
+        this->actor.velocity.y = -25.0f;
+        this->actor.terminalVelocity = -25.0f;
+        this->actor.gravity = -5.0f;
+    }
+}
+
+void EnAni_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+}
+
+void EnAni_SetText(EnAni* this, GlobalContext* globalCtx, u16 textId) {
+    s16 diffAngle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    this->actor.textId = textId;
+    if ((this->stateFlags & ANI_STATE_SUFFERING) || ABS_ALT(diffAngle) <= 0x4300) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            func_800B8614(&this->actor, globalCtx, 120.0f);
+        }
+    }
+}
+
+void EnAni_IdleStanding(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnAni_Talk(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Message_GetState(&globalCtx->msgCtx) == 2 && globalCtx->msgCtx.unk11F04 == 0x6DE) {
+        this->actionFunc = EnAni_IdleInPain;
+    }
+}
+
+void EnAni_IdleInPain(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        this->actionFunc = EnAni_Talk;
+    } else {
+        // telling you not to take his rupees you knocked from the tree
+        EnAni_SetText(this, globalCtx, 0x6DE);
+    }
+}
+
+void EnAni_FallOverInPain(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->blinkFunc = EnAni_WaitForEyeOpen;
+        this->actionFunc = EnAni_IdleInPain;
+        Animation_Change(&this->skelAnime, &gAniHoldingFootPainWrithingAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniHoldingFootPainWrithingAnim), 0, 0.0f);
+    }
+}
+
+void EnAni_LandOnFoot(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actionFunc = EnAni_FallOverInPain;
+        Animation_Change(&this->skelAnime, &gAniFallOverHoldingFootAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniFallOverHoldingFootAnim), 2, 0.0f);
+    }
+}
+
+void EnAni_FallToGround(EnAni* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 quakeValue;
 
-extern ColliderCylinderInit D_80968670;
-extern InitChainEntry D_8096869C[];
+    if (this->actor.bgCheckFlags & BGCHECK_CHECK_WALL) {
+        this->actor.flags &= ~ACTOR_FLAG_10;
```

Comment:
this define is not meant for this value. bgcheckflags are an actor-facing system which is seperate but very related to bgcheck. `BGCHECK_CHECK_WALL` is a bcc flag internal to bgcheck.

There are no macros for bgCheckFlags in either game as far as I know